### PR TITLE
Docker swarm: add filtering of services

### LINKED ISSUE #8073: Swarm Service Discovery: Smarter, more efficient retrieval of data.
As it stands, all objects [tasks,nodes,networks,..] are pulled from the swarm API on interval N for each service discovery block.
In a relatively small swarm (10 nodes), I have ~400 containers with 9 service discovery blocks. For each iteration, currently 120+ MB is pulled over the line.

Rather than GET'ting all objects from the API each time and filtering the output on the Prometheus side through relabeling, we could be a lot smarter and more efficient by not pulling the (majority) of irrelevant data in the first place. The Swarm API has built-in support for (basic) filtering operations.

I actually noticed this, due to load spikes on one of the manager nodes. In my case, I run a "Caddy" container on a manager node which exposes the Swarm API, through mounting the socket and in turn exposing it to the Prometheus backend.

### PR DESCRIPTION
**Pull request by roidelapluie**
Add filters on all docker swarm roles (nodes, tasks and services).

Fix #8073 

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -577,6 +577,12 @@ role: <string>
 # tasks and services that don't have published ports.
 [ port: <int> | default = 80 ]
 
+# Optional filters to limit the discovery process to a subset of available
```
Comment (brian-brazil) [inline docs/configuration/configuration.md:580]: The usual paragraph about only using this for performance reasons should be added.

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -577,6 +577,23 @@ role: <string>
...
+# resources.
+# The avabilable filters are listed in the upstream documentation:
```
Comment (ArthurSens) [inline docs/configuration/configuration.md:582]:
```suggestion
# The available filters are listed in the upstream documentation:
```
:stuck_out_tongue:

**Comment by roidelapluie** [comment]
@brian-brazil ping, can we have this in?

File: docs/configuration/configuration.md (hunk 1 above)
Comment (brian-brazil) [reply docs/configuration/configuration.md:580]: This comment is still unaddressed.

File: docs/configuration/configuration.md [hunk 3]
```diff
@@ -577,6 +577,23 @@ role: <string>
...
 [ port: <int> | default = 80 ]
 
+# Optional filters to limit the discovery process to a subset of available
+# resources.
+# The available filters are listed in the upstream documentation:
+# Services: https://docs.docker.com/engine/api/v1.40/#operation/ServiceList
+# Tasks: https://docs.docker.com/engine/api/v1.40/#operation/TaskList
+# Nodes: https://docs.docker.com/engine/api/v1.40/#operation/NodeList
+
+# Note: When making decision about using filters make sure that this is the best
```
Comment (brian-brazil) [inline docs/configuration/configuration.md:587]: I wouldn't blindly copy this from k8s without knowing it's an issue.

**Comment by roidelapluie** [comment]
I have updated the comment

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**